_posts/2019-12-11-reproducible-firmware-builds.md

```diff
@@ -328,6 +328,28 @@ freertos_kernel/portable/MemMang/heap_1.c
 
 Great! We no longer have absolute paths in the binary.
 
+As an alternative, the compiler flag
+[`-ffile-prefix-map=old=new`](https://gcc.gnu.org/onlinedocs/gcc-12.1.0/gcc/Overall-Options.html#index-ffile-prefix-map)
+performs a substitution for path prefix segments:
+
+```bash
+$ arm-none-eabi-gcc -ffile-prefix-map=/private/tmp=. -E /private/tmp/test.c
+[...]
+const char *my_file_name = "./test.c";
+```
+
+> Note: the `-ffile-prefix-map=old=new` was first added in GCC 8.1, although
+> apparently without enough excitement to deserve a mention in the [release
+> notes](https://gcc.gnu.org/gcc-8/changes.html)!<br/><br/>
+> This flag sets all `f*-prefix-map` settings, so it applies to debug info as
+> well as macro (`__FILE__`) substitutions. Note that it does NOT as of yet get
+> forwarded to `as` when invoked from the GCC frontend, so be sure to add
+> `-Wa,--debug-prefix-map=old=new` if that applies.<br/><br/>
```

Comment:
Should this always be set? Would the previous instructions have had things covered?

Reply:
Updated the note a smidge, it is probably a mistake in GCC (still required as of 12.1), since `-fdebug-prefix-map` does forward the arg to `as`.